openadapt/config.py

```diff
@@ -128,3 +142,17 @@ def filter_log_messages(data):
         "Cannot pickle Objective-C objects",
     ]
     return not any(msg in data["message"] for msg in messages_to_ignore)
+
+
+def set_db_url(db_fname):
+    """Set the database URL based on the given database file name.
+
+    Args:
+        db_fname (str): The database file name.
+    """
+    global DB_FNAME, DB_FPATH, DB_URL
+    DB_FNAME = db_fname
+    DB_FPATH = ROOT_DIRPATH / DB_FNAME
+    DB_URL = f"sqlite:///{DB_FPATH}"
+    logger.info(f"{DB_URL=}")
+    os.environ["DB_FNAME"] = db_fname
```

Comment:
@Mustaballer can you please clarify why this line is necessary?

Reply:
This line is necessary because the way we overwrite the constants is either from DEFAULTS or from the os environment variables(It looks for the os environment first). The DEFAULT for `DB_FNAME` is `openadapt.db`. If this line were omitted, the code would use the default value from the environment variable, which is also openadapt.db.

```python
def getenv_fallback(var_name):
    """Get the value of an environment variable with fallback to default.

    Args:
        var_name (str): The name of the environment variable.

    Returns:
        str: The value of the environment variable or the default value if not found.

    Raises:
        ValueError: If the environment variable is not defined.
    """
    rval = os.getenv(var_name) or _DEFAULTS.get(var_name) # <-- this line here
    if rval is None:
        raise ValueError(f"{var_name=} not defined")
    return rval


load_dotenv()

for key in _DEFAULTS:
    val = getenv_fallback(key)
    locals()[key] = val
```

Reply:
Hi @Mustaballer , I'm still not following. `getenv_fallback` is only called once per config param, at startup. After calling `set_db_url` it will not be called.